tabpfn_client/config.py

```diff
@@ -52,32 +52,35 @@ def init(use_server=True):
                 )
 
             # prompt for login / register
-            PromptAgent.prompt_and_set_token(user_auth_handler)
+            print("prompt_and_set_token")
+            PromptAgent.prompt_and_set_token()
 
         # Print new greeting messages. If there are no new messages, nothing will be printed.
         PromptAgent.prompt_retrieved_greeting_messages(
-            user_auth_handler.retrieve_greeting_messages()
+            UserAuthenticationClient.retrieve_greeting_messages()
         )
 
-        g_tabpfn_config.use_server = True
-        g_tabpfn_config.user_auth_handler = user_auth_handler
-        g_tabpfn_config.inference_handler = InferenceClient(service_client)
-
+        Config.use_server = True
+        Config.is_initialized = True
     else:
         raise RuntimeError("Local inference is not supported yet.")
-        # g_tabpfn_config.use_server = False
-
-    g_tabpfn_config.is_initialized = True
 
 
 def reset():
-    # reset config
-    global g_tabpfn_config
-    g_tabpfn_config = TabPFNConfig()
-
+    Config.is_initialized = False
     # reset user auth handler
-    if g_tabpfn_config.use_server:
-        g_tabpfn_config.user_auth_handler.reset_cache()
+    if Config.use_server:
+        UserAuthenticationClient.reset_cache()
 
     # remove cache dir
     shutil.rmtree(CACHE_DIR, ignore_errors=True)
+
+
+def get_access_token() -> str:
+    init()
+    return ServiceClient.get_access_token()
+
+
+def set_access_token(access_token: str):
```

Comment:
if I use `set_access_token` with a wrong token, I get a very cryptic error:
```
RuntimeError: Fail to call upload_train_set with error: 500, reason: Internal Server Error and text: Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/jwt/api_jws.py", line 258, in _load
    signing_input, crypto_segment = jwt.rsplit(b".", 1)
ValueError: not enough values to unpack (expected 2, got 1)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.10/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
  File "/usr/local/lib/python3.10/site-packages/fastapi/routing.py", line 291, in app
    solved_result = await solve_dependencies(
  File "/usr/local/lib/python3.10/site-packages/fastapi/dependencies/utils.py", line 640, in solve_dependencies
    solved = await run_in_threadpool(call, **solved_result.values)
  File "/usr/local/lib/python3.10/site-packages/starlette/concurrency.py", line 39, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/usr/local/lib/python3.10/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 2441, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 943, in run
    result = context.run(func, *args)
  File "/code/tabpfn-server/app/dependencies.py", line 115, in get_current_user
    status, user = user_auth.get_user_by_token(token)
  File "/code/tabpfn-server/app/services/user_auth_service.py", line 265, in get_user_by_token
    token_data = self._decode_token(token)
  File "/code/tabpfn-server/app/services/user_auth_service.py", line 279, in _decode_token
    return self._jwt_token_handler.decode_token(token)
  File "/code/tabpfn-server/app/services/user_auth_service.py", line 58, in decode_token
    return jwt.decode(token, self._JWT_SECRET, algorithms=self._JWT_ALGO)
  File "/usr/local/lib/python3.10/site-packages/jwt/api_jwt.py", line 211, in decode
    decoded = self.decode_complete(
  File "/usr/local/lib/python3.10/site-packages/jwt/api_jwt.py", line 152, in decode_complete
    decoded = api_jws.decode_complete(
  File "/usr/local/lib/python3.10/site-packages/jwt/api_jws.py", line 199, in decode_complete
    payload, signing_input, header, signature = self._load(jwt)
  File "/usr/local/lib/python3.10/site-packages/jwt/api_jws.py", line 261, in _load
    raise DecodeError("Not enough segments") from err
jwt.exceptions.DecodeError: Not enough segments
```

Reply:
this should be fixed in a server-side pr